serverless/serverless-azure-functions
# refactor: Move function.json logic into new FunctionBuildService

## What did you implement:

- New `FunctionBuildService` class containing logic for building `function.json` files.


## How did you implement it:

- Logic for building `function.json` files was an awkward fit inside the `Utils` class. This moves the logic into a new class, the `FunctionBuildService`. Because it inherits from the `BaseService`, this will provide access to the service configuration (precursor to adding Python & other runtime support)
- Added some additional interfaces and simplification of code
- There will likely be additional simplification and refactoring within this class in the future, but made minimal changes for now



## How can we verify it:

- Easiest way is to run `sls offline build` and inspect the `function.json` files. The `offlineService` and `packageService` unit tests helped verify the implementation



## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

       **Validate via `npm test`**
       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_

**_Is this ready for review?:_** NO  
**_Is it a breaking change?:_** NO
